dashboard/pages/screening/callbacks.py

```diff
@@ -654,94 +682,78 @@ def register_callbacks(elements, file_storage):
+        Input("activation-formula-dropdown", "value"),
+    )(functools.partial(on_additional_options_change))
+
+    callback(
+        Output("compounds-data-table", "children"),
         Output("z-score-plot", "figure"),
-        Output("activation-plot", "figure"),
-        Output("inhibition-plot", "figure"),
+        Output("feature-plot", "figure"),
         Output("z-score-min-input", "value"),
         Output("z-score-max-input", "value"),
-        Output("activation-min-input", "value"),
-        Output("activation-max-input", "value"),
-        Output("inhibition-min-input", "value"),
-        Output("inhibition-max-input", "value"),
+        Output("feature-min-input", "value"),
+        Output("feature-max-input", "value"),
         Output("compounds-data-subtitle", "children"),
+        Output("tab-feature-header", "children"),
+        Output("filter-radio", "options"),
         Input(elements["STAGES_STORE"], "data"),
         State("user-uuid", "data"),
         State("z-slider-value", "data"),
+        State("activation-inhibition-screening-options", "data"),
     )(functools.partial(on_summary_entry, file_storage=file_storage))
-
+    # Z-SCORE
     callback(
         Output("z-score-plot", "figure", allow_duplicate=True),
         Output("z-score-button", "disabled"),
         Input("z-score-min-input", "value"),
         Input("z-score-max-input", "value"),
         State("z-score-plot", "figure"),
         prevent_initial_call=True,
-    )(functools.partial(on_range_update, key="Z-SCORE"))
+    )(functools.partial(on_range_update))
     callback(
         Output("z-score-plot", "figure", allow_duplicate=True),
         Input("z-score-button", "n_clicks"),
         State("user-uuid", "data"),
         State("z-score-min-input", "value"),
         State("z-score-max-input", "value"),
         State("z-score-plot", "figure"),
+        State("activation-inhibition-screening-options", "data"),
         prevent_initial_call=True,
     )(
         functools.partial(
-            on_apply_button_click, file_storage=file_storage, key="Z-SCORE"
+            on_apply_button_click, key="Z-SCORE", file_storage=file_storage
         )
     )
+    # ACTIVATION/INHIBITION
     callback(
-        Output("activation-plot", "figure", allow_duplicate=True),
-        Output("activation-button", "disabled"),
-        Input("activation-min-input", "value"),
-        Input("activation-max-input", "value"),
-        State("activation-plot", "figure"),
+        Output("feature-plot", "figure", allow_duplicate=True),
+        Output("feature-button", "disabled"),
+        Input("feature-min-input", "value"),
+        Input("feature-max-input", "value"),
+        State("feature-plot", "figure"),
+        State("activation-inhibition-screening-options", "data"),
         prevent_initial_call=True,
-    )(functools.partial(on_range_update, key="% ACTIVATION"))
+    )(functools.partial(on_range_update))
```

Comment:
partial not needed here (no args are filled), it's sufficient to pass the `on_range_update` func itself